velox/functions/prestosql/DecimalArithmetic.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/lib/LambdaFunctionUtil.h"
+#include "velox/type/DecimalUtils.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+template <
+    typename R /* Result Type */,
+    typename A /* Argument1 */,
+    typename B /* Argument2 */,
+    typename Operation /* Arithmetic operation */>
+class DecimalBaseFunction : public exec::VectorFunction {
+ public:
+  DecimalBaseFunction(const uint8_t aRescale, const uint8_t bRescale)
```

Comment:
I started making some comments on the subtraction PR.
No need of const here.

Reply:
Done.